sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,388 @@
+allocated by a second API. If the second API modifies that memory, then this
+process would have to be repeated in the opposite direction in order for the
+first API to see the changes made to that memory.
+
+This extension enables copy-free sharing of SYCL allocated device memory with
+external APIs.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_MEMORY_EXPORT` to one of the values defined in the
+table below. Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Rev | Description
+|1   | Initial draft of the proposal
+|======================
+
+=== Querying device support
+
+We provide the following device aspect to query for support of exporting memory.
+
+[frame="none",options="header"]
+|======================
+|Device descriptor |Description
+|`aspect::ext_oneapi_exportable_device_mem` | Indicates if the device supports
+the allocation of exportable linear memory and exporting that memory to an
+interoperable handle.
+|======================
+
+=== External Memory Resource Handle Types [[external_mem_res_handles]]
+
+This extension provides an enum `external_mem_handle_type` that defines several
+external memory resource handle types that can be used as interoperable
+handles to import SYCL allocated memory into external APIs.
+
+[_Note:_ Not all of the handle types defined in this enum may be supported for
+exporting memory by the implementation. Currently, the {dpcpp} implementation
+only supports exporting memory with the `opaque_fd` and `win32_nt_handle` handle
+types. This enum is shared with the memory import functionality defined in the
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc[sycl_ext_oneapi_bindless_images]
+extension, where more handle types may be supported for importing memory into
+SYCL.
+_{endnote}_]
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+// External memory resource handle types.
+enum class external_mem_handle_type {
+  opaque_fd = 0,
+  win32_nt_handle = 1,
+  win32_nt_dx12_resource = 2,
+  dma_buf = 3,
+  win32_nt_dx11_resource = 4,
+};
+
+template <external_mem_handle_type ExternalMemHandleType>
+struct exported_mem;
+
+template <>
+struct exported_mem<external_mem_handle_type::opaque_fd> {
+  using type = int;
+};
+
+template <>
+struct exported_mem<external_mem_handle_type::win32_nt_handle> {
+  using type = void *;
+};
+
+template <external_mem_handle_type ExternalMemHandleType>
+using exported_mem_t = typename exported_mem<ExternalMemHandleType>::type;
+
```

Comment:
The `exported_mem` and `exported_mem_t` types should be removed now that they aren't used below.